modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/RuleItem.es.js

```diff
@@ -86,6 +93,23 @@ export default function RuleItem({rule, toggleRulesEditorVisibility}) {
 		},
 	];
 
+	const getLabel = (fieldName) => {
+		const field = getDataDefinitionField(dataDefinition, fieldName);
+		let dataLayoutLabel;
+		let dataDefinitionLabel;
+
+		if (!field?.customProperties?.labelAtStructureLevel) {
+			dataLayoutLabel =
+				dataLayoutFields?.[fieldName]?.label?.[editingLanguageId];
+		}
+
+		if (field) {
+			dataDefinitionLabel = getFieldLabel(dataDefinition, fieldName);
+		}
+
+		return dataLayoutLabel || dataDefinitionLabel;
+	};
+
```

Comment:
Hey @MateusCastro, could you improve that function? I didn't test my function but I think you can do something like:

```javascript
function getLabel({dataDefinition, dataLayout}, editingLanguageId, fieldName) {
    const field = getDataDefinitionField(dataDefinition, fieldName);

    if (!field?.customProperties?.labelAtStructureLevel) {
        return dataLayout?.dataLayoutFields?.[fieldName]?.label?.[editingLanguageId];
    }

    return field?.label?.[editingLanguageId];
}
```